### PR TITLE
logging: add template for log configuration

### PR DESCRIPTION
**Pull request by nordic-krch**
Added Kconfig template file to be used when log configuration
in a module is added. Added example usage of kconfig template
in logger sample.

It is using new kconfig feature added in: #9797 (issue #9761)

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

**Comment by nordic-krch** [comment]
fyi @Olivier-ProGlove @jhedberg @jukkar see the example how log configuration can be established in each module:
https://github.com/zephyrproject-rtos/zephyr/blob/1d8b49eb592a13e966949d0603b9e36d4bf2ea12/samples/subsys/logging/logger/Kconfig#L13

**Review by jukkar — CHANGES_REQUESTED** [review]
Could this be split to two commits? The first one would add the template, and the second one would add the sample that shows how to use it.

**Comment by nordic-krch** [comment]
@jukkar done.

**Review by jukkar — APPROVED** [review]
LGTM

**Review by carlescufi — APPROVED** [review]
Looks good, Will add @ulfalizer to review too.

**Comment by jukkar** [comment]
I have been porting this to networking logging code and was wondering if something like this would be possible (ping @ulfalizer) :

Kconfig:
```
module=FOOBAR
module-str=FOOBAR component
module-help=Enables FOOBAR debug messages.
source "Kconfig.log_level_template"
```

Kconfig.log_level_template:
```
choice
	prompt "Log level for $(module-str)"
	default $(module)_LOG_LEVEL_WRN
	help
	  $(module-help)
```

Current code outputs the help text just plain "$(module-help)" instead of expected help text.

File: samples/subsys/logging/logger/Kconfig [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+mainmenu "Logger sample application"
+
+menu "Application configuration"
+
+module=SAMPLE_MODULE
+module-str=Sample module
```
Comment (ulfalizer) [inline samples/subsys/logging/logger/Kconfig]: Would look less cramped with spaces around the `=`.

**Review by ulfalizer — APPROVED** [review]

**Comment by ulfalizer** [comment]
> Current code outputs the help text just plain "$(module-help)" instead of expected help text.

No macro expansion is done in help texts, so not currently possible. I wonder if it would break stuff in case help texts talk about Make variables, though that's unlikely in Zephyr.

Do you think making a reference to the prompt would be possible in the meantime, or making it generic ("enables some-feature for the subsystem" or the like)?

**Review by nashif — COMMENTED** [review]
do we have documentation how this template can be used and will be used?

**Comment by nordic-krch** [comment]
good point. it is work updating the documentation.

**Comment by carlescufi** [comment]
This is great. Just a minor comment, should we settle on an extension for Kconfig templates? I would perhaps suggest “.ktemplate”

**Comment by nordic-krch** [comment]
@carlescufi would you propose renaming file to **log_config.ktemplate** ?

**Comment by carlescufi** [comment]
>@carlescufi would you propose renaming file to log_config.ktemplate ?

Actually looking at what we do today:
`Kconfig.defconfig.nrf51822_QFAC`
I think the consistent naming scheme would be:
`Kconfig.template.logging`

**Comment by nordic-krch** [comment]
@carlescufi maybe `Kconfig.template.log_config` since in future we might want to add another template (all must be in separate files) so `Kconfig.template.logging` is to wide. Don't you think?

**Comment by carlescufi** [comment]
@nordic-krch

Sounds good to me.

**Review by nashif — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
doc change looks OK.

**Pull request merged**